File: arrays/caesar/caesar.py
def verify_key(key):
  for k in key:
    K = ord(k)
    if(K < ord('0') or K > ord('9')) : return False
  return True
def get_ciphertext(plaintext, key):
  if (plaintext == ''): return 'plaintext is empty!'
  ciphertext = 'ciphertext: '
  i, j, KEY= [1, 0.1, 0]
  for k in key:
    KEY += (ord(k) - ord('0')) * i
    i *= 0.1
    j *= 10
  KEY *= j
  for char in plaintext:
    CHAR = ord(char.upper())
    ciphertext += chr(int(ord('A') +(CHAR - ord('A') + KEY)%26)) if((ord(char) >= ord('A')) and (ord(char) <= ord('Z'))) else  (chr(int(ord('a') +(ord(char) - ord('a') + KEY)%26)) if((ord(char) >= ord('a')) and (ord(char) <= ord('z'))) else char)
  return ciphertext


def encrypt():
  key = input('key: ')
  if(verify_key(key)):
    plaintext = input('plaintext: ')
    print(get_ciphertext(plaintext, key))
  else:
    print('key MUST be a POSITIVE INTEGER!')
    encrypt()
  
encrypt()
  
  

File: arrays/caesar/caesar.cpp
#include <iostream>
bool verifyKey(std::string key);
std::string ciphertext(std::string plaintext, std::string key);
int main(int argc, char* argv[])
{
  if(argc != 2 || !verifyKey(argv[1]))
  {
    std::cout << "usage: ./caesar key {key must be a positive integer}";
    return 1;
  }
  else
  {
    std::cout << "plaintext:  ";
    std::string plaintext = "";
    std::getline(std::cin, plaintext);
    std::cout << ciphertext(plaintext, argv[1]) << "\n";
    return 0;
  }
}

bool verifyKey(std::string key)
{
  for(char c : key)
  {
    if(c < '0' || c > '9')
    {
      return false;
    }
  }
  return true;
}

std::string ciphertext(std::string plaintext, std::string key){
  if(plaintext == "") return "plaintext is empty!";
  std::string ciphertext = "ciphertext: ";
  int Key = 0;
  float j = 1, i = 0.1, l = 0;
  for(char k : key)
  {
    l += (k - '0') * j;
    j *= 0.1;
    i *= 10;
  }
  l *= i;
  Key = static_cast<int>(l);
  for(char c : plaintext)
  {
    if(c >= 'A' && c <= 'Z') ciphertext += ('A' + (c - 'A' + Key)%26);
    else if(c >= 'a' && c <= 'z') ciphertext += ('a' + (c - 'a' + Key)%26);
    else ciphertext += c;
  }
  return ciphertext;
}